### PR TITLE
Refactoring NNPTPZ1_8_10_2022

### PR DESCRIPTION
**Pull request by Tomdokx**
Refactored code

st60982.

### CODE REVIEW

File: NNPTPZ1/Program.cs [hunk 1]
```diff
@@ -1,331 +1,272 @@
...
 namespace NNPTPZ1
 {
-    /// <summary>
-    /// This program should produce Newton fractals.
     /// See more at: https://en.wikipedia.org/wiki/Newton_fractal
-    /// </summary>
-    class Program
+    class NewtonFractalProducer
     {
         static void Main(string[] args)
```
Comment (rodi0878) [inline NNPTPZ1/Program.cs:11]: method is too complex

File: NNPTPZ1/Program.cs [hunk 2]
```diff
@@ -1,331 +1,272 @@
...
-                    //Console.WriteLine(ox);
+                    if (Point.RealNumber == 0)
+                        Point.RealNumber = 0.0001;
+                    if (Point.ImaginaryUnit == 0)
+                        Point.ImaginaryUnit = 0.0001f;
 
-                    // find solution of equation using newton's iteration
-                    float it = 0;
-                    for (int q = 0; q< 30; q++)
+                    float iteration = 0;
```
Comment (rodi0878) [inline NNPTPZ1/Program.cs]: why float?

File: NNPTPZ1/Program.cs [hunk 3]
```diff
@@ -1,331 +1,272 @@
...
-            //    }
-            //}
-
-                    bmp.Save(output ?? "../../../out.png");
-            //Console.ReadKey();
+            bitmapToPaint.Save(output ?? "../../../out.png");
         }
     }
 
     namespace Mathematics
```
Comment (rodi0878) [inline NNPTPZ1/Program.cs]: separate classes to separate files

**Review by rodi0878 — CHANGES_REQUESTED** [review]
please see mentioned problems and provide update

**Comment by Tomdokx** [comment]
More refactored code, problems should be solved.

File: NNPTPZ1/Mathematics/NewtonFractal.cs [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            PathToSaveBitmap = null;
+            if (inicialisationParameters.Length == 7)
+            {
+                PathToSaveBitmap = inicialisationParameters[pointerForArguments];
+            }
+
+            BitmapToDraw = new Bitmap(WidthOfBitmap, HeightOfBitmap);
+        }
+
+        public void Draw()
```
Comment (rodi0878) [inline NNPTPZ1/Mathematics/NewtonFractal.cs:86]: Method is too complex

**Review by rodi0878 — CHANGES_REQUESTED** [review]

**Comment by rodi0878** [comment]
Accepted

**Pull request closed without merging**